src/radical/pilot/agent/scheduler/base.py

```diff
@@ -450,11 +453,14 @@ def _change_slot_states(self, slots, new_state):
     #
     # NOTE: any scheduler implementation which uses a different nodelist
     #       structure MUST overload this method.
-    def slot_status(self, msg=None):
+    def slot_status(self, msg=None, uid=None):
```

Comment:
`uid` should be a part of `msg` - no need to add extra inputs

Reply:
`slot_status` is called with high frequency, and I wanted to avoid (somewhat costly) string expansion when the string might not be needed.  Also avoids code clutter to do string expansion in a single place.